File: backend/parking-api/src/main/java/com/chatchatabc/api/application/dto/user/UserDTO.java
package com.chatchatabc.api.application.dto.user;

import com.chatchatabc.api.application.dto.role.RoleDTO;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.Set;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserDTO implements Serializable {
    private String id;
    private String email;
    private String username;
    private String phone;
    private String firstName;
    private String lastName;
    private Set<RoleDTO> roles;

    // TODO: Add the other dates?
}

File: parking-api/src/main/kotlin/com/chatchatabc/parking/api/application/dto/user/UserDTO.kt
package com.chatchatabc.parking.api.application.dto.user

data class UserPhoneLoginRequest(
    val phone: String,
    val username: String?
)

data class UserProfileUpdateRequest(
    val username: String?,
    val phone: String?,
    val email: String?,
    val firstName: String?,
    val lastName: String?
)

data class UserVerifyOTPRequest(
    val phone: String,
    val otp: String
)